pkg/services/notification/handler.go

```diff
@@ -24,7 +24,18 @@ import (
 func (s *Server) SetServiceConfig(ctx context.Context, req *pb.ServiceConfig) (*pb.SetServiceConfigResponse, error) {
 	err := ValidateSetServiceConfigParams(ctx, req)
 	if err != nil {
-		return nil, err
+		return nil, gerr.NewWithDetail(ctx, gerr.InvalidArgument, err, gerr.ErrorValidateEmailService)
+	}
+
+	respValidate, err := s.ValidateEmailService(ctx, req)
+	if err != nil {
+		logger.Errorf(ctx, "Failed to set service config, %+v.", err)
+		return nil, gerr.NewWithDetail(ctx, gerr.InvalidArgument, err, gerr.ErrorValidateEmailService)
```

Comment:
The err is already gerr.NewWithDetail(ctx, gerr.InvalidArgument, err, gerr.ErrorValidateEmailService)

No need to gerr.NewWithDetail again.